plugins/dialog/plugin.js

```diff
@@ -954,23 +946,41 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 		 */
 		layout: function() {
 			var el = this.parts.dialog;
-			var dialogSize = this.getSize();
-			var win = CKEDITOR.document.getWindow(),
-					viewSize = win.getViewPaneSize();
 
-			var posX = ( viewSize.width - dialogSize.width ) / 2,
+			if ( !this._.moved && ( !CKEDITOR.env.ie || CKEDITOR.env.edge ) ) {
```

Comment:
I wonder why you using `flex` for dialog wrapper if the view has not been moved. Isn't better to just recalculate dialog position for the first load? It seems like without this condition dialog works better on iOS (maybe accident, requires additional testing). Not sure if `flex` should be used as a standalone CSS property (outside flexbox layout), especially that it's used with `table` layout - which isn't great also.

Reply:
I'm using `flex` because it is easiest way to center element inside another element with pure CSS, so we don't need to have JS for that. 

Using `display: flex` on wrapper will make all its children flex items, unless they have defined `position: fixed` or `position: absolute`. Even if dialog outermost element is using table layout to position its children it can be flex item to position itself inside flex wrapper.